drivers/dac/dac_mcp4728.c

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2021 Grinn
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_mcp4728
+
+#include <zephyr.h>
+#include <drivers/dac.h>
+#include <drivers/i2c.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(dac_mcp4728, CONFIG_DAC_LOG_LEVEL);
+
+#define MCP4728_MAX_CHANNEL			4
+#define MCP4728_RESOLUTION			0xC
+#define MCP4728_DAC_MAX_VAL			((2 << (MCP4728_RESOLUTION - 1)) - 1)
```

Comment:
More easy to understand:
```suggestion
#define MCP4728_DAC_MAX_VAL			((1U << MCP4728_RESOLUTION) - 1U)
```